doc/nrf/config_and_build/companion_components.rst

```diff
@@ -0,0 +1,71 @@
+.. _companion_components:
+
+Using companion components
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |NCS| provides several companion components.
+Typically, a companion component is independent from the application and is included in your project as a separate firmware image.
+
+Companion components are intended to be used as-is.
+While you may modify them or change the configuration, they are only tested for the default configuration set.
```

Comment:
What does `change the configuration` mean? Does following the integration instruction count as changing the configuration?

In the case of the `flash_companion` some configuration from the user in device tree is needed to integrate the component, so maybe we could phrase it as such:

```suggestion
Integration of some companion components may require changing the configuration. Refer to the component documentation for tested and supported integration instructions.
```

Reply:
Yes, we can probably tune the sentence. I had kconfig in my mind but we could try to describe it better. My intent is to say that the user should operate in the confinement of what we test. 

We can state that each companion component defines it but I'm not sure if that is the case today.

Reply:
How I read the edits from @wbober : the component C is meant to be used  without changes in application A. You can change the component (C+1), but we don't guarantee it will work, because only C is tested.

Your suggestion is different to me: you speak of integration, so using the example above: to integrate component C one needs to modify application A to A+1.

So you're talking about two different things here. Your comments @e-rk make more sense in the section about adding components.

Reply:
Kept the line here as-is. Used Rafal's suggestion in the "Adding" section below.

Reply:
To clarify a bit from my side, the user has application A and wants to integrate component C.

To integrate the component with the application, depending on the component, the user might have to change the configuration of the application from A to A+1. Same thing may apply to the component, which might need to be modified from C to C+1.

The component should have a constrained integration instruction, so that going from C to C+1 when following it does not reduce our guarantees.

There may be some instances, where that might not be possible. For example, in flash companion the user has to do 2 things:
* create a partition in NVM and place the companion there
* enable an external flash driver appropriate for the device the user has on board. In Nordic we can only test against the external flash device we have on the DK.

If the user only changes NVM partition, that should not reduce our guarantees. If the user changes the external flash driver, then that a more complicated matter.